adit/selective_transfer/tests/test_download.py

```
@@ -86,3 +102,96 @@ async def wait_until_ready(self):
+):
+    # The download re-checks server access at fetch time. A server the user's active
+    # group cannot access as a source must yield NotFound and never touch the PACS.
+    user = user_with_download_rights
+    server = DicomServerFactory.create()  # not granted -> inaccessible
+
+    operator_cls = mocker.patch(
+        "adit.selective_transfer.utils.dicom_downloader.DicomOperator"
+    )
+
+    downloader = DicomDownloader(server_id=str(server.pk))
+    with pytest.raises(NotFound, match="Invalid DICOM server."):
```

Ruff (0.15.20)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)